### PR TITLE
feat(scoring): complete scoring intelligence slice

### PR DESCRIPTION
**Pull request by ebarti**
## Summary

This PR implements a PR-sized scoring intelligence slice against `main` after PR #57 merged.

- Makes scoring criteria-aware and constraint-aware: saved score criteria, target criteria, profile preference fields, fit band, confidence, eligibility, hard blockers, missing/transferable/matched signals, criteria snapshots, and score trace metadata are now represented in the Python scoring aggregate and persistence.
- Wires score corrections end to end: API contract/client/route, latest-score version write, `ScoreCorrected` event, projection/read-model updates, web optimistic mutation with rollback, and a compact job drawer correction control.
- Adds local scoring eval and feedback surfaces: synthetic fixtures and metrics for parse validity, band accuracy, blocker precision/recall, nDCG@k, correction agreement, plus transparent feedback signals from existing corrections and job action events.
- Adds governance and safety rails: non-sensitive scoring span metadata, persisted trace/correction history, QA eval gate docs, and applicant-side/employer-side boundary notes.

## Validation

Local only; GitHub Actions are paused per instruction.

- `uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_score_use_cases.py workers/automation/tests/test_scorer.py` -> 17 passed
- `uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_scoring_eval*.py` -> 3 passed
- `corepack pnpm api:test` -> 6 files passed, 90 tests passed
- `corepack pnpm --filter @jobhunter/web test -- src/contexts/scoring src/views/jobs` -> 83 files passed, 435 tests passed; Vitest emitted the existing `--localstorage-file` warning
- `corepack pnpm api:check` -> passed
- `corepack pnpm web:check` -> passed

Additional local checks:

- `uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_score_repository.py` -> 13 passed
- `uv --project workers/automation run --extra dev ruff check workers/automation/src/jobhunter/domain/scoring workers/automation/src/jobhunter/infrastructure/scoring workers/automation/src/jobhunter/scoring workers/automation/tests/test_score_use_cases.py workers/automation/tests/test_scorer.py workers/automation/tests/test_scoring_eval.py workers/automation/tests/test_scoring_eval_feedback.py workers/automation/tests/test_score_repository.py` -> passed
- `git diff --check` -> passed
- Final retarget check after PR #57 merge: `git diff --check origin/main...HEAD` -> passed

## Notes / Risks

- This intentionally keeps `FitScore` at 1..10. Eligibility blockers are persisted and surfaced separately instead of changing downstream score semantics.
- Feedback personalization is transparent and local, but still lightweight: corrections and action events produce inspectable signals/ranking adjustments; no embedding-backed or trained ranker is introduced in this slice.
- The drawer correction control is compact and covered by unit/a11y-focused suites, but I did not run a browser/Playwright manual path because the requested validation was local unit/typecheck and CI is paused.

### CODE REVIEW

File: workers/automation/src/jobhunter/domain/scoring/services.py [hunk 1]
```diff
@@ -193,3 +376,47 @@ class EligibilityChecker:
     def is_eligible(self, fit_score: FitScore, criteria: ScoringCriteria) -> bool:
         """True iff the candidate clears the configured ``min_fit_score`` cut."""
         return fit_score.value >= criteria.min_fit_score
```
Comment (ebarti) [inline workers/automation/src/jobhunter/domain/scoring/services.py:387]: PR Reviewer - Codex

Severity: High
Gate: Fail
Evidence: This predicate still returns only `fit_score.value >= criteria.min_fit_score`, while the production selectors for tailoring/cover/apply also gate only on `_EFFECTIVE_FIT_SCORE >= ?`: `workers/automation/src/jobhunter/database.py:2052-2060`, `workers/automation/src/jobhunter/pipeline/runner.py:1154-1174`, and `workers/automation/src/jobhunter/apply/launcher.py:230-252`. The new regression test even proves a job can persist `fit_score == 9` with `eligibility.status == "blocked"` (`workers/automation/tests/test_score_use_cases.py:391-401`). The plan requires downstream gating to use both score and `eligibility.status != blocked`, and auto-apply must require no hard blockers.
Reproduction: Reasoned from code

Hard blockers are represented but not enforced on the downstream queues, so a blocked job with a high score can still be tailored, covered, and auto-applied.

Required fix: Carry latest score eligibility into the queue selectors/read helpers and exclude `eligibility.status = 'blocked'` (or any non-empty hard blockers) from tailoring, cover generation, and apply acquisition; add regression coverage for a high-score blocked job not being selected.

File: workers/automation/src/jobhunter/infrastructure/scoring/feedback.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+def collect_feedback_signals(conn: sqlite3.Connection) -> tuple[ScoringFeedbackSignal, ...]:
+    """Collect transparent feedback signals from local scoring/action tables."""
+
+    signals: list[ScoringFeedbackSignal] = []
+    signals.extend(_correction_signals(conn))
+    signals.extend(_action_signals(conn))
+    return tuple(signals)
+
+
+def rank_jobs_with_feedback(
```
Comment (ebarti) [inline workers/automation/src/jobhunter/infrastructure/scoring/feedback.py:53]: PR Reviewer - Codex

Severity: High
Gate: Fail
Evidence: The new feedback helpers are exported and unit-tested, but I found no production caller in `workers/automation/src/jobhunter/scoring/scorer.py`, `workers/automation/src/jobhunter/pipeline/runner.py`, `workers/automation/src/jobhunter/database.py`, or `workers/automation/src/jobhunter/scoring/activities.py` (`rg "collect_feedback_signals|rank_jobs_with_feedback|FeedbackRankedJob|ScoringFeedbackSignal|feedback"` returned no runtime hits there). The only direct use is the new test `workers/automation/tests/test_scoring_eval_feedback.py:6,52-60`.
Reproduction: Reasoned from code

This adds a feedback scoring library, but it does not wire feedback personalization into scoring, ranking, queue selection, or any user-visible/API path. For a completion slice, that is modeled behavior rather than live behavior.

Required fix: Put the feedback adjustment on an actual production path, for example when producing/ranking scored candidates or selecting jobs for downstream stages, and add a test that proves corrections or skip/apply actions change the real ordering/selection without hiding the underlying evidence.

File: apps/web/src/contexts/scoring/components/ScoreCorrectionControl.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const [correctedScore, setCorrectedScore] = useState(String(currentScore ?? 7));
+  const [reason, setReason] = useState("");
+  const mutation = useCorrectScoreMutation();
+  const disabled = mutation.isPending || !reason.trim();
+
+  const submit = (event: FormEvent) => {
+    event.preventDefault();
+    mutation.mutate({
+      jobId,
+      correctedScore: Number(correctedScore),
```
Comment (ebarti) [inline apps/web/src/contexts/scoring/components/ScoreCorrectionControl.tsx]: PR Reviewer - Codex

Severity: Medium
Gate: Pass
Evidence: The control disables submit only when the reason is blank (`line 15`) and then submits `Number(correctedScore)` here. Because the number input is not `required`, clearing it leaves the form valid; `Number("")` becomes `0`, so the optimistic mutation patches the query cache to `fitScore: 0` before the API rejects it. That transient UI state violates the preserved 1..10 FitScore semantics.
Reproduction: Reasoned from code

The API schema correctly rejects scores outside 1..10, but the web path can still optimistically render an invalid score before rollback.

Required fix: Validate the score in the control before calling `mutate` and disable/guard submit unless it is an integer from 1 through 10; add a component or hook test for the empty/out-of-range path.

**Review by ebarti — COMMENTED** [review]
PR Reviewer - Codex

Severity: High
Gate: Fail
Evidence: The scoring completion goal requires hard blockers to participate in downstream safety gates, but the changed code only persists/displays blockers and the production selectors still gate by fit score.
Reproduction: Reasoned from code

Inline findings below are blocking for this slice.

File: workers/automation/src/jobhunter/domain/scoring/services.py (hunk 1 above)
Comment (ebarti) [reply workers/automation/src/jobhunter/domain/scoring/services.py:387]: PR Reviewer - Codex

Severity: High
Gate: Fail
Evidence: The fixer commit now filters the batch selectors and `acquire_job`, but the single-job flow still bypasses those gates. In `workers/automation/src/jobhunter/pipeline/runner.py:1741-1763`, `apply_jobs` loads or creates a score and only copies `fit_score` into the job dict; it never checks `existing_score.breakdown.eligibility` or `outcome.score.breakdown.eligibility`. It then immediately enters tailoring at `workers/automation/src/jobhunter/pipeline/runner.py:1768-1773` and cover generation at `workers/automation/src/jobhunter/pipeline/runner.py:1797-1817`. The later `apply_main(target_url=..., min_score=0)` now goes through the launcher blocker gate, so apply acquisition is fixed, but blocked jobs can still enter targeted tailoring and cover generation.
Reproduction: Reasoned from code

The blocker gate remains incomplete for the single-job `job ... --tailor` / `apply_jobs(do_tailor=True)` path: a high-score blocked job can still generate tailored resume and cover-letter artifacts before apply rejects it.

Required fix: Before tailoring in `apply_jobs`, load/check the latest `JobScore` eligibility (or call `EligibilityChecker.is_eligible(..., eligibility=score.breakdown.eligibility)`) and stop the tailor/cover flow when `eligibility.status == "blocked"` or hard blockers are non-empty; add a regression for the targeted tailor path with a high-score blocked job.

Comment (ebarti) [reply workers/automation/src/jobhunter/domain/scoring/services.py:387]: PR Fixer - Codex

Fixed in `bba281db4408a4e448b204155e210de3c4b50961`. The single-job `run_single_job` path now preserves the latest `JobScore` aggregate after load/create and exits before tailoring or cover generation when `score.breakdown.eligibility.status == "blocked"` or hard blockers are present. Added `test_single_job_tailor_rejects_blocked_candidate` to prove a targeted `job ... --tailor`-style call does not invoke tailoring or cover generation for a high-score blocked job.

Verification passed locally:
- `uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_apply_regressions.py workers/automation/tests/test_score_queue_selectors.py workers/automation/tests/test_apply_queue_selectors.py` -> 35 passed
- `uv --project workers/automation run --extra dev pytest -q workers/automation/tests/test_score_use_cases.py workers/automation/tests/test_scorer.py workers/automation/tests/test_scoring_eval*.py` -> 21 passed
- `uv --project workers/automation run --extra dev ruff check workers/automation/src/jobhunter/pipeline/runner.py workers/automation/tests/test_apply_regressions.py` -> passed
- `corepack pnpm api:test` -> 6 files passed, 90 tests passed
- `corepack pnpm --filter @jobhunter/web test -- src/contexts/scoring src/views/jobs` -> 83 files passed, 437 tests passed, existing `--localstorage-file` warning emitted
- `corepack pnpm api:check` -> passed
- `corepack pnpm web:check` -> passed
- `git diff --check` -> passed
